File: Backend/NavNexus.Domain/Entities/Evidence.cs
namespace NavNexus.Domain.Entities;

public class Evidence
    {
        public string Id { get; set; }                   // unique evidence ID
        public string SourceId { get; set; }             // ID document gốc
        public string SourceName { get; set; }           // tên file gốc
        public string ChunkId { get; set; }              // ID chunk liên quan
        public string Text { get; set; }                 // nội dung evidence
        public int Page { get; set; }                    // trang trong document
        public float Confidence { get; set; }            // độ tin cậy
        public DateTime CreatedAt { get; set; }          // thời gian tạo
        public string Language { get; set; }   = "ENG"  ;       // ngôn ngữ
        public string SourceLanguage { get; set; }  = "ENG"  ;     // ngôn ngữ gốc
        public string HierarchyPath { get; set; } = String.Empty;       // đường dẫn trong tài liệu

        public List<string> Concepts { get; set; } = new List<string>(); // danh sách concepts

        public List<string> KeyClaims { get; set; } = new List<string>(); // các claim quan trọng
        public List<string> QuestionsRaised { get; set; } = new List<string>(); // câu hỏi liên quan
        public float EvidenceStrength { get; set; }      // độ tin cậy của node

        public Evidence()
        {
            Id = Guid.NewGuid().ToString();
            SourceId = string.Empty;
            SourceName = string.Empty;
            ChunkId = string.Empty;
            Text = string.Empty;
            Page = 0;
            Confidence = 0.0f;
            CreatedAt = DateTime.UtcNow;
            Language = "ENG";
            SourceLanguage = "ENG";
            HierarchyPath = string.Empty;
            EvidenceStrength = 0.0f;
        }
    }

File: RabbitMQ/src/model/Evidence.py
from __future__ import annotations
from dataclasses import dataclass, field
from typing import List
from datetime import datetime, timezone
import uuid

@dataclass
class Evidence:
    Id: str = field(default_factory=lambda: str(uuid.uuid4()))
    SourceId: str = ""
    SourceName: str = ""
    ChunkId: str = ""
    Text: str = ""
    Page: int = 0
    Confidence: float = 0.0
    CreatedAt: datetime = field(default_factory=lambda: datetime.now(timezone.utc))
    Language: str = "ENG"
    SourceLanguage: str = "ENG"
    HierarchyPath: str = ""
    Concepts: List[str] = field(default_factory=list)
    KeyClaims: List[str] = field(default_factory=list)
    QuestionsRaised: List[str] = field(default_factory=list)
    EvidenceStrength: float = 0.0

    # Position tracking fields
    StartPos: int = 0  # Character position where text starts in original document
    EndPos: int = 0    # Character position where text ends in original document
    ChunkIndex: int = 0  # Sequential chunk number
    HasMore: bool = False  # Whether there are more chunks after this one
    OverlapLength: int = 0  # Length of overlap with previous chunk
